tutorials/polkadot-sdk/build-a-blockchain/add-trusted-nodes.md

```diff
@@ -0,0 +1,313 @@
+- It simulates a more realistic blockchain environment where participants don't share key information
+- It helps you understand the process of key generation, which is a fundamental skill in blockchain operations
+
+To understand more about the different signing algorithms used in this tutorial (sr25519 and ed25519), check [Keypairs and Signing](https://wiki.polkadot.network/docs/learn-cryptography#keypairs-and-signing){target=\_blank}. To learn more about the different types of keys used, refer to the [Keys](https://wiki.polkadot.network/docs/learn-cryptography#keys){target=\_blank} section in the Polkadot Wiki.
+
+### Key Generation Options
+
+There are several ways you can generate keys. The available methods are:
+
+- `solochain-template-node` [key](TODO:update-path) subcommand - the most straightforward method for developers working directly with the node is to use the integrated key generation feature. Using the `key` subcommand, you can generate keys directly from your node's command line interface. This method ensures compatibility with your chain and is convenient for quick setup and testing
+- [subkey](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/bin/utils/subkey){target=\_blank} - it is a powerful standalone utility specifically designed for Polkadot SDK-based chains. It offers advanced options for key generation, including support for different key types such as `ed25519` and `sr25519`. This tool allows fine-grained control over the key generation process
+- Third-party key generation utilities - various tools developed by the community
+
+### Generate Local Keys with the Node Template
+
+Best practices for key generation:
+
+- Whenever possible, use an air-gapped computer, meaning never connected to the internet, when generating keys for a production blockchain
+- If an air-gapped computer is not an option, disconnect from the internet before generating keys for any public or private blockchain not under your control
+
+For this tutorial, however, you'll use the `solochain-template-node` command-line options to generate random keys locally while remaining connected to the internet. This method is suitable for learning and testing purposes.
+
+Follow these steps to generate your keys:
+
+1. Navigate to the root directory where you compiled the node template
+
+2. Generate a random secret phrase and Sr25519 keys. Enter a password when prompted:
+
+    ```bash
+    ./target/release/solochain-template-node key generate \
+    --scheme Sr25519 \
+    --password-interactive
+    ```
+
+    The command will output information about the generated keys similar to the following:
+
+    --8<-- 'code/tutorials/polkadot-sdk/build-a-blockchain/add-trusted-nodes/key-sr25519-1.html'
+
+    Note the Sr25519 public key for the account (SS58 format). This key will be used for producing blocks with `aura`. In this example, the Sr25519 public key for the account is `5HMhkSHpD4XcibjbU9ZiGemLpnsTUzLsG5JhQJQEcxp3KJaW`.
```

Comment:
is aura referring to code here, otherwise I feel like we should spell it like Aura